crates/net/network/src/session/mod.rs

```diff
@@ -805,6 +809,13 @@ async fn authenticate_stream(
             }
         }
     };
+
+    let mut eth_stream = eth_stream;
```

Comment:
nit: We should just do `let (mut eth_stream, their_status) = match ...` on the line above

Reply:
🤦‍♂️ yes my b